src/bench/inputfetcher.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2025-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <bench/data/block413567.raw.h>
+#include <coins.h>
+#include <common/system.h>
+#include <inputfetcher.h>
+#include <primitives/block.h>
+#include <serialize.h>
+#include <streams.h>
+#include <util/time.h>
+
+static constexpr auto DELAY{2ms};
+
+//! Simulates a DB by adding a delay when calling GetCoin
+class DelayedCoinsView : public CCoinsView
```

Comment:
I personally would favor going for simpler code as opposed to going for theoretically better encapsulation, similarly to current `inputfetcher_tests.cpp`:
```C++
struct DelayedCoinsView : CCoinsView
```